util/jsonutil/jsonutil_test.go

```diff
@@ -240,3 +245,118 @@ func TestTryExtractErrorMessage(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateEncoder(t *testing.T) {
+	formatted := `{
+  "properties": {
+    "string": "Blanks spaces in between words to not be removed if compacted",
+    "integer": 5,
+    "string_array": [
+      "string array elem one",
+      "string array elem two"
+    ]
+  }
+}`
+	compacted := `{"properties":{"string":"Blanks spaces in between words to not be removed if compacted","integer":5,"string_array":["string array elem one","string array elem two"]}}`
+
+	type testCase struct {
+		desc string
+		test func(t *testing.T)
+	}
+	testGroup := []struct {
+		desc  string
+		tests []testCase
+	}{
+		{
+			desc: "Extension registered",
+			tests: []testCase{
+				{
+					desc: "JSON inside a string field is passed to Marshal(), don't expect the output to be compacted",
+					test: func(t *testing.T) {
+						jsoniter.RegisterExtension(&RawMessageExtension{})
+						out, err := Marshal(formatted)
+						assert.NoError(t, err)
+						assert.NotEqual(t, compacted, string(out))
+					},
+				},
+				{
+					desc: "json.RawMessage is passed to Marshal(), expect inner JSON blob to be line-break-free, tab-free, spaces only found inside strings, and compacted into one line",
+					test: func(t *testing.T) {
+						jsoniter.RegisterExtension(&RawMessageExtension{})
+						out, err := Marshal(json.RawMessage(formatted))
+						assert.NoError(t, err)
+						assert.Equal(t, compacted, string(out))
+					},
+				},
+			},
+		},
+		{
+			desc: "Extension not registered, json.RawMessage won't get compacted",
+			tests: []testCase{
+				{
+					desc: "json.RawMessage not cleared of line breaks, tabs, nor compacted into one line",
+					test: func(t *testing.T) {
+						jsoniter.RegisterExtension(&RawMessageExtension{})
+						out, err := Marshal(json.RawMessage(formatted))
+						assert.NoError(t, err)
+						assert.Equal(t, compacted, string(out))
+					},
+				},
+			},
+		},
+	}
+
+	for _, group := range testGroup {
+		for _, tc := range group.tests {
+			t.Run(fmt.Sprintf("%s - %s", group.desc, tc.desc), tc.test)
+		}
+	}
```

Comment:
I would call this an anti-pattern since there is no common code between the tests. It would be better to write this without the loop, just three separate `t.Run` calls. Same for `TestEncode`.

Reply:
Simplified and removed common code. `TestEncode` now calls two `t.Run()` calls